snippets/deepFreeze.md

```diff
@@ -0,0 +1,38 @@
+### deepFreeze
+
+Deep freezes an object.
+
+Calls `Object.freeze(obj)` recursively on all unfrozen properties of obj that are `typeof` function or object.
+
+```js
+const deepFreeze = obj => {
+  Object.freeze(obj);
+
+  Object.getOwnPropertyNames(obj).forEach(function(prop) {
+    if (
+      obj.hasOwnProperty(prop) &&
+      obj[prop] !== null &&
+      (typeof obj[prop] === 'object' || typeof obj[prop] === 'function') &&
```

Comment:
Why not use
```js
obj[prop] instanceof Object
```

Reply:
This would require a change to the snippet description since you wouldn't be using typeof anymore

Reply:
I agree that `instanceof` is probably better and shorter. Let's give it a shot.